pkg/clients/redis/redis.go

```diff
@@ -0,0 +1,229 @@
+	if err != nil {
+		return nil, err
+	}
+	c := &client{redisCli: redisCli}
+	return c, nil
+}
+
+func (c *client) DescribeDBInstance(id string) (*DBInstance, error) {
+	request := aliredis.CreateDescribeInstancesRequest()
+	request.Scheme = httpsScheme
+
+	request.InstanceIds = id
+
+	response, err := c.redisCli.DescribeInstances(request)
+	if err != nil {
+		return nil, err
+	}
+	if len(response.Instances.KVStoreInstance) == 0 {
+		return nil, ErrDBInstanceNotFound
+	}
+	rsp := response.Instances.KVStoreInstance[0]
+	in := &DBInstance{
+		ID:     rsp.InstanceId,
+		Status: rsp.InstanceStatus,
+	}
+
+	return in, nil
+}
+
+func (c *client) CreateDBInstance(req *CreateRedisInstanceRequest) (*DBInstance, error) {
+	pw, err := password.Generate()
+	if err != nil {
+		return nil, err
+	}
+	request := aliredis.CreateCreateInstanceRequest()
+	request.Scheme = "https"
+
+	request.InstanceName = req.Name
+	request.EngineVersion = req.EngineVersion
+	request.InstanceClass = req.InstanceClass
+	request.InstanceType = req.Engine
+	request.ReadTimeout = 60 * time.Second
+	request.Password = pw
+	request.ChargeType = "PostPaid"
+
+	resp, err := c.redisCli.CreateInstance(request)
+	if err != nil {
+		return nil, err
+	}
+
+	return &DBInstance{
+		ID: resp.InstanceId,
+		Endpoint: &v1alpha1.Endpoint{
+			Address: resp.ConnectionDomain,
+			Port:    strconv.Itoa(resp.Port),
+		},
+	}, nil
+}
+
+func (c *client) CreateAccount(id, user, pw string) error {
+	request := aliredis.CreateCreateAccountRequest()
+	request.Scheme = "https"
+	request.InstanceId = id
+	request.AccountName = user
+	request.AccountPassword = pw
+	request.ReadTimeout = 60 * time.Second
+
+	_, err := c.redisCli.CreateAccount(request)
+	return err
+}
+
+func (c *client) DeleteDBInstance(id string) error {
+	request := aliredis.CreateDeleteInstanceRequest()
+	request.Scheme = "https"
+
+	request.InstanceId = id
+
+	_, err := c.redisCli.DeleteInstance(request)
+	return err
+}
+
+// GenerateObservation is used to produce v1alpha1.RedisInstanceObservation from
+// redis.DBInstance.
+func GenerateObservation(db *DBInstance) redisv1alpha1.RedisInstanceObservation {
+	return redisv1alpha1.RedisInstanceObservation{
+		DBInstanceStatus: db.Status,
+		DBInstanceID:     db.ID,
+	}
+}
+
+// MakeCreateDBInstanceRequest generates CreateDBInstanceRequest
+func MakeCreateDBInstanceRequest(name string, p *redisv1alpha1.RedisInstanceParameters) *CreateRedisInstanceRequest {
+	if p.Engine == "" {
+		p.Engine = "Redis"
+	}
+
+	return &CreateRedisInstanceRequest{
+		Name:           name,
+		Engine:         p.Engine,
+		EngineVersion:  p.EngineVersion,
+		SecurityIPList: p.SecurityIPList,
+		InstanceClass:  p.DBInstanceClass,
+		Port:           p.DBInstancePort,
+	}
+}
+
+// IsErrorNotFound helper function to test for ErrCodeDBInstanceNotFoundFault error
+func IsErrorNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+	// If instance already remove from console.  should ignore when delete instance
```

Comment:
```suggestion
	// If the instance is already removed, errors should be ignored when deleting it.
```

Just wanna to remove `from console`.